.travis.yml

```diff
@@ -25,6 +25,24 @@ matrix:
           packages: [ 'libstdc++6', 'libstdc++-5-dev' ]
     - os: osx
       osx_image: xcode8.3
+    # clang-tidy specific job
+    - os: linux
+      sudo: false
+      addons:
+        apt:
+          sources: [ 'ubuntu-toolchain-r-test' ]
+          packages: [ 'libstdc++6', 'libstdc++-5-dev' ]
+      script:
+        - make tidy
```

Comment:
Heads up that this might not fail even if there are problems in the code. When playing around with clang-tidy in node-cpp-skel, I ran into a bug in `clang-tidy` whereby even when it reports warnings/errors it does not return a `non-zero` error code. Super frustrating! Maybe I am missing something. However in the meantime I noticed that often `clang-tidy` is able to modify the source code automatically (not always mind you). So in `node-cpp-skel` I made the clang-tidy run in the same job as clang-format. Then, if clang-tidy finds errors it can fix, it will fix them, and then the job will fail because the clang-format script fails if it detects changes in git.